### ISSUE TITLE
Create `SkeletonProvider` component

### ISSUE THREAD

**Issue by AlexFrazer** [chakra-ui/chakra-ui#2130]
# Feature request

## Is your feature request related to a problem? Please describe.

Just repetition in one component that could be alleviated. Suppose I have a component that looks something similar to this:

![image](https://user-images.githubusercontent.com/1350392/94727375-7c52df80-0313-11eb-9884-0b670924bb0e.png)

Now, I want to make the major areas surrounded in skeletons

![image](https://user-images.githubusercontent.com/1350392/94727677-ebc8cf00-0313-11eb-878d-84f646c6982a.png)

Here's the problem: whenever I have a loading state, I have to pass it into _each individual skeleton_, i.e., `<Skeleton isLoading />`. On this page I have roughly 4-5 skeletons. This makes it quite tedious to refactor

[Here is a code sandbox link](https://codesandbox.io/s/chakra-home-page-forked-0e10n?file=/src/App.tsx)

## Describe the solution you'd like

ideally, a `SkeletonProvider` where I can pass in the `isLoading` and the `Skeleton`, `SkeletonText`, and `SkeletonCircle` will read from that context.

For individual components, it would be possible to pass in a `isLoading` that overrides the context. i.e.

```
<SkeletonProvider isLoading>
  <Skeleton isLoading={isCardImagesLoading}>
    <Image src="blah" />
  </Skeleton>
  <Skeleton>
    <Heading>Fox</Skeleton>
  </Skeleton>
  <SkeletonText>
    <Text>Lorem ipsum</Text>
  </SkeletonText>
  <Skeleton>
    <Footer />
  </Skeleton>
</SkeletonProvider>
```

## Describe alternatives you've considered

Basically just passing in the `isLoading` to every component or manually implementing the Skeleton provider on my own app.

**Comment by segunadebayo** [comment]
Hello @AlexFrazer,

I'm a bit hesitant to add this component to Chakra UI. Kindly implement it on your end and share your solution so others can learn from it as well. We could add it as a guide in the docs.

Thank you.

**Comment by abhijit-hota** [comment]
What is the preferred method of showing a Skeleton for an `<Image/>` component?
